### PR TITLE
sort-imports autofix

### PR DESCRIPTION
**Pull request by jwbay**
I'm kind of upset by how easy this was.

### CODE REVIEW

File: rules/sortImportsRule.ts [hunk 1]
```diff
@@ -8,34 +9,61 @@ export class Rule extends Lint.Rules.AbstractRule {
...
+				const sortableLine = importDeclaration.getText(sf)
+					.toLowerCase()
+					.replace('import {', 'import +');
+				sortableLinesToImports.set(sortableLine, importDeclaration);
+				return sortableLine;
+			});
+
+			const sortedLines = unsortedLines.slice().sort();
+
+			for (let i = 0; i < unsortedLines.length; i += 1) {
```
Comment (bookman25) [inline rules/sortImportsRule.ts]: Interesting choice here `i +=1` instead of `i++`, any reason?

File: rules/sortImportsRule.ts [hunk 2]
```diff
@@ -8,34 +9,61 @@ export class Rule extends Lint.Rules.AbstractRule {
...
-			for (let i = 0; i < importLines.length; i += 1) {
-				if (importLinesForComparison[i] !== sortedImportLines[i]) {
+	public visitSourceFile(sf: ts.SourceFile) {
+		for (const importGroup of this.getImportGroups(sf)) {
+			const sortableLinesToImports = new Map<string, ts.ImportDeclaration>();
+			const unsortedLines = importGroup.map(importDeclaration => {
+				// opening brace is below alphanumeric characters char-code-wise, but want named imports above defaults
+				// + comes directly after *, so swap with that
+				const sortableLine = importDeclaration.getText(sf)
+					.toLowerCase()
```
Comment (bookman25) [inline rules/sortImportsRule.ts]: So this only handles sorting sans case?

File: rules/sortImportsRule.ts (hunk 1 above)
Comment (bookman25) [inline rules/sortImportsRule.ts]: Interesting choice here `i +=1` instead of `i++`, any reason?

File: rules/sortImportsRule.ts (hunk 2 above)
Comment (jwbay) [inline rules/sortImportsRule.ts]: Yeah, it's always been that way (for this rule). I think A-z is easier to scan through to find a line than A-Za-z

File: rules/sortImportsRule.ts (hunk 1 above)
Comment (jwbay) [inline rules/sortImportsRule.ts]: I have a lint rule on that forbids it because += 1 is more explicit and can't be confused with ++i

**Pull request merged**